res/controllers/Traktor-Kontrol-S3-hid-scripts.js

```diff
@@ -1112,10 +1112,17 @@ TraktorS3.Deck = class {
     }
 
     loadTrackHandler(field) {
-        if (this.shiftPressed) {
-            engine.setValue(this.activeChannel, "eject", field.value);
+        // If the library is selected, load the track based on which encoder was
+        // pressed. Otherwise just do whatever the default action is for the
+        // focused menu or widget.
+        if (engine.getValue("[Library]", "focused_widget") === 3) {
+            if (this.shiftPressed) {
+                engine.setValue(this.activeChannel, "eject", field.value);
+            } else {
+                engine.setValue(this.activeChannel, "LoadSelectedTrack", field.value);
+            }
```

Comment:
FWIW, for the S4 which has also two track encoders, I use LoadSelectedTrack only if the deck's not playing, else GoToItem (load to next not playing deck). This saves me to switch the encoder when I'm currently browsing on the right side.

Reply:
That's a good idea too, I'll make a follow up; thanks!